common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java

```diff
@@ -0,0 +1,139 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.migration;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.google.gson.Gson;
+import com.microsoft.identity.common.internal.cache.ADALTokenCacheItem;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class AdalMigrationAdapterDeserializationTest {
+
+    private static final String AUTHORITY = "https://login.microsoftonline.com/common";
+    private static final String CLIENT_ID = "a_client_id";
+    private static final String TENANT_ID = "a_tenant_id";
+    public static final String CACHE_KEY = "a_cache_key";
+
+    private interface DeserializationValidator {
+        void onDeserializationFinished(Map<String, ADALTokenCacheItem> result);
+    }
+
+    private AdalMigrationAdapter mMigrationAdapter;
+
+    private Map<String, String> mDeserializationInput;
+    private DeserializationValidator mValidator;
+
+    @Parameterized.Parameters
+    public static Iterable<Object[]> testParams() {
+        return Arrays.asList(
+                new Object[]{
+                        getAdalCacheItemInput(),
+                        getDeserializationValidator()
+                },
+                new Object[]{
+                        getMalformedCacheItemInput(),
+                        getDeserializationValidator()
+                }
+        );
+    }
```

Comment:
You know I love parameterized tests, but the question I would ask is that with only 2 permutations here is it clearer to just have the two methods?

Reply:
Originally had exactly that, and expanded upon it:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/957/commits/7ad92ac9b6279811ca9e37851070a63b4fac06c4

So reverting this is trivial, your call on which you want

Reply:
I'm not concerned with it, really.  An name parameter would be nice, but I doubt that your class arguments give useful toString values.

Reply:
Added a name param via the 0th param: 81f02b1f98dd4f78c46c65940bd42826ad2f607f